### PR TITLE
[AIRFLOW-2100] Fix Broken Documentation Links

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2100

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
- Replaced `Pythonhosted` documentation links with `Apache Docs` as agreed and documented here: https://cwiki.apache.org/confluence/display/AIRFLOW/Building+and+deploying+the+docs?show-miniview

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Docs changed

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 [![PyPI version](https://badge.fury.io/py/apache-airflow.svg)](https://badge.fury.io/py/apache-airflow)
 [![Build Status](https://travis-ci.org/apache/incubator-airflow.svg?branch=master)](https://travis-ci.org/apache/incubator-airflow)
 [![Coverage Status](https://img.shields.io/codecov/c/github/apache/incubator-airflow/master.svg)](https://codecov.io/github/apache/incubator-airflow?branch=master)
-[![Documentation](https://img.shields.io/badge/docs-pythonhosted-blue.svg)](http://pythonhosted.org/airflow/)
+[![Documentation Status](https://readthedocs.org/projects/airflow/badge/?version=latest)](http://airflow.readthedocs.io/en/latest/?badge=latest)
```
Comment (Fokko) [inline README.md]: Please make it `https://`.

File: README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ makes it easy to visualize pipelines running in production,
 monitor progress, and troubleshoot issues when needed.
 
 ## Getting started
-Please visit the Airflow Platform documentation for help with [installing Airflow](http://pythonhosted.org/airflow/installation.html), getting a [quick start](http://pythonhosted.org/airflow/start.html), or a more complete [tutorial](http://pythonhosted.org/airflow/tutorial.html).
+Please visit the Airflow Platform documentation for help with [installing Airflow](http://airflow.incubator.apache.org/installation.html), getting a [quick start](http://airflow.incubator.apache.org/start.html), or a more complete [tutorial](http://airflow.incubator.apache.org/tutorial.html).
```
Comment (Fokko) [inline README.md]: Please make it `https://` and I think it would be wise to explicitly mention that these docs are the latest and not the latest release.

**Review by Fokko — CHANGES_REQUESTED** [review]
With a few changes I think it is good to merge this now, and then get another place where  the docs of Airflow 1.9 are hosted.

File: README.md (hunk 1 above)
Comment (kaxil) [reply README.md]: Done

**Comment by kaxil** [comment]
@Fokko 

* **Readthedocs** links point to the latest GitHub version of Airflow. [link](https://airflow.readthedocs.io/en/latest/) 
* **Apache Doc** links point to latest stable release i.e. 1.9 for now. [link](https://airflow.incubator.apache.org)

**Comment by Fokko** [comment]
@kaxil Thanks for fixing!

**Pull request closed without merging**